internal/shakex4/f1600x4_amd64_src.go

```diff
@@ -0,0 +1,148 @@
+// +build ignore
+
+// AVX2 fourway parallelized KeccaK-f[1600].
+
+package main
+
+import (
+	. "github.com/mmcloughlin/avo/build"
+	. "github.com/mmcloughlin/avo/operand"
+)
+
+func main() {
+	ConstraintExpr("amd64")
+
+	TEXT("f1600x4AVX2", 0, "func(state *[100]uint64, rc *[24]uint64)")
+
+	// Pragma("noescape")  // XXX
+	// We want state to be 16 byte alligned.  Go only guarantees an
+	// (array of) uint64 to be alligned on 8 bytes except if it's heap
+	// allocated.  Thus we do not add the noescape pragma to ensure that
+	// state is heap allocated and thus 16 byte alligned.
+
+	state_ptr := Load(Param("state"), GP64())
+	state := func(offset int) Op {
+		return Mem{Base: state_ptr, Disp: 32 * offset}
+	}
+
+	rc_ptr := Load(Param("rc"), GP64())
```

Comment:
If you don't mind duplicating the round constants you could also define them as a `DATA` section.

https://github.com/mmcloughlin/avo/tree/master/examples/data